src/cleanlab_codex/validator.py

```diff
@@ -0,0 +1,303 @@
+        self,
+        query: str,
+        context: str,
+        response: str,
+        prompt: Optional[str] = None,
+        form_prompt: Optional[Callable[[str, str], str]] = None,
+    ) -> dict[str, Any]:
+        """Evaluate whether the AI-generated response is bad, and if so, request an alternate expert answer.
+        If no expert answer is available, this query is still logged for SMEs to answer.
+
+        Args:
+            query (str): The user query that was used to generate the response.
+            context (str): The context that was retrieved from the RAG Knowledge Base and used to generate the response.
+            response (str): A reponse from your LLM/RAG system.
+            prompt (str, optional): Optional prompt representing the actual inputs (combining query, context, and system instructions into one string) to the LLM that generated the response.
+            form_prompt (Callable[[str, str], str], optional): Optional function to format the prompt based on query and context. Cannot be provided together with prompt, provide one or the other. This function should take query and context as parameters and return a formatted prompt string. If not provided, a default prompt formatter will be used. To include a system prompt or any other special instructions for your LLM, incorporate them directly in your custom form_prompt() function definition.
+
+        Returns:
+            dict[str, Any]: A dictionary containing:
+                - 'expert_answer': Alternate SME-provided answer from Codex if the response was flagged as bad and an answer was found in the Codex Project, or None otherwise.
+                - 'is_bad_response': True if the response is flagged as potentially bad, False otherwise. When True, a Codex lookup is performed, which logs this query into the Codex Project for SMEs to answer.
+                - Additional keys from a [`ThresholdedTrustworthyRAGScore`](/cleanlab_codex/types/validator/#class-thresholdedtrustworthyragscore) dictionary: each corresponds to a [TrustworthyRAG](/tlm/api/python/utils.rag/#class-trustworthyrag) evaluation metric, and points to the score for this evaluation as well as a boolean `is_bad` flagging whether the score falls below the corresponding threshold.
+        """
+        loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(loop)
+        try:
+            expert_answer, scores, is_bad_response = loop.run_until_complete(
+                self._validate_async(query, context, response, prompt, form_prompt)
+            )
+
+            if is_bad_response and expert_answer is None:
+                # TODO: Make this async as well in the future (right now it takes 8% of the time)
+                with ThreadPoolExecutor(max_workers=1) as executor:
+                    executor.submit(
+                        self._project._sdk_client.projects.entries.add_question,
+                        self._project._id, question=query
+                    ).result()
```

Comment:
Before merge, a SWE needs to confirm there isn’t some simple latency gain left on the table in this line.  Or that at least, the proper backend Eng tickets are in place to reduce this line's latency.

This line should ideally be near instant, any backend operations shouldn’t block it client side (eg. actually organizing how the new query gets logged in the DB shouldn't block the client).